src/cargo/util/semver_ext.rs

```diff
@@ -49,22 +51,48 @@ impl OptVersionReq {
                     cmp.op == Op::Exact && cmp.minor.is_some() && cmp.patch.is_some()
                 }
             }
+            OptVersionReq::Locked(..) => true,
```

Comment:
Is the `is_exact` method still used anywhere? If sow why?

Reply:
IIUC, for `cargo install` to decide whether to skip registry update or not. Though I found that some other duplicated logic laying there. 😂
https://github.com/rust-lang/cargo/pull/9847/files#diff-085d4f28bd6e49633caa467300082e94527f840dd487242b13bad2882a25451fR678